app/src/main/kotlin/com/saltedge/authenticator/features/connections/list/ConnectionsListCollector.kt

```diff
@@ -28,41 +28,71 @@ import com.saltedge.authenticator.core.model.isActive
 import com.saltedge.authenticator.core.tools.toDateTime
 import com.saltedge.authenticator.features.connections.common.ConnectionItem
 import com.saltedge.authenticator.models.Connection
+import com.saltedge.authenticator.models.location.DeviceLocationManagerAbs
 import com.saltedge.authenticator.tools.ResId
 import com.saltedge.authenticator.tools.toDateFormatString
 
-fun Connection.convertConnectionToViewModel(context: Context): ConnectionItem {
+fun Connection.convertConnectionToViewModel(context: Context, deviceLocationManager: DeviceLocationManagerAbs): ConnectionItem {
+    val locationPermissionsIsGranted: Boolean = deviceLocationManager.locationPermissionsGranted(context)
+    val shouldGrantAccess = shouldRequestPermission(geolocationRequired = this.geolocationRequired, locationPermissionsAreGranted = locationPermissionsIsGranted)
     return ConnectionItem(
         guid = this.guid,
         connectionId = this.id,
         name = this.name,
-        statusDescription = getConnectionStatusDescription(context = context, connection = this),
-        statusDescriptionColorRes = getConnectionStatusColor(connection = this),
+        statusDescription = getConnectionStatusDescription(context = context, connection = this, grantAccessToLocationData = shouldGrantAccess),
+        statusDescriptionColorRes = getConnectionStatusColor(connection = this, grantAccessToLocationData = shouldGrantAccess),
         logoUrl = this.logoUrl,
         isActive = this.isActive(),
         isChecked = false,
         apiVersion = this.apiVersion,
-        email = this.supportEmail
+        email = this.supportEmail,
+        locationPermissionRequired = shouldGrantAccess
     )
 }
 
-fun List<Connection>.convertConnectionsToViewModels(context: Context): List<ConnectionItem> {
-    return this.map { connection -> connection.convertConnectionToViewModel(context) }
+fun List<Connection>.convertConnectionsToViewModels(context: Context, locationManager: DeviceLocationManagerAbs): List<ConnectionItem> {
+    return this.map { connection -> connection.convertConnectionToViewModel(context, locationManager) }
 }
 
-private fun getConnectionStatusDescription(context: Context, connection: Connection): String {
+/**
+ * Should request permission if geolocationRequired is null or false
```

Comment:
Wrong description. Why we need do ask permission when geolocationRequired is false?